desc/compute/data_index.py

```diff
@@ -148,10 +148,30 @@ def _decorator(func):
             flag = False
             for base_class, superclasses in _class_inheritance.items():
                 if p in superclasses or p == base_class:
-                    if name in data_index[base_class]:
+                    if p != base_class:
+                        # track if the compute definition was from using
+                        # a superclass
+                        defined_from_superclass = True
+                    else:
+                        #
+                        defined_from_superclass = False
+                    if (
+                        name in data_index[base_class]
+                        and not data_index[base_class][name]["defined_from_superclass"]
+                    ):
+                        # only throw an error if we are trying to redefine the same thing twice
+                        # but do not throw error if the definition was through the superclass,
+                        # as this means we defined a second compute function for the subclass
+                        # specifically, so we want to allow that to overwrite the previous
+                        # definition.
```

Comment:
Another way to do this would be to add `parameterization` to `d` above (to keep track of the parameterization that defines the method), then if you get a second definition of a given function you check `_class_inheritance` and prefer subclasses over superclasses.

something like 
```python
if p in superclasses or p == base_class:
    if name in data_index[p] and superclass in data_index[p]['parameterization']:
        # overriding parent class
        data_index[p][name] = d
    else:
        raise ValueError(...)
```

Reply:
I believe it but the logic is not coming to me in the couple minutes I spent trying this way. If someone else wants to refactor it then go ahead

Reply:
This idea seems better if we can get it working. Would that avoid the issue with the order the functions are defined in?